### PR TITLE
Tests should accept a specific pipeline version (git repo) as a parameter

### PR DESCRIPTION
**Pull request by kptdobe**
PR for #1242 

With this you can run: 

```
env HLX_CUSTOM_PIPELINE="https://github.com/adobe/helix-pipeline.git#winston-compatability" npm run test
```

to run the tests with a custom pipeline.

This introduces a HLX_CUSTOM_PIPELINE env variable (or customPipeline) which would be the npm dependency to install - potential value could be `@adobe/helix-pipeline`, `@adobe/helix-pipeline@<version>` or `https://github.com/adobe/helix-pipeline.git#branch` (anything npm install accepts).

This also fixes the env variables interferences between tests (env variables get lost after first call of `clearHelixEnv` and are not available to following tests).

### CODE REVIEW

File: src/build.js [hunk 1]
```diff
@@ -28,6 +28,14 @@ module.exports = function build() {
     builder: (yargs) => {
       yargsBuild(yargs);
```
Comment (tripodsan) [inline src/build.js]: why not add to `yargsBuild` ? it is needed also in all other commands, like `up` and `package` and `deploy` ...

File: src/builder/ModuleHelper.js [hunk 2]
```diff
@@ -130,7 +130,7 @@ class ModuleHelper {
     }
   }
 
-  async installModule(name) {
+  async installModule(name, installer) {
```
Comment (tripodsan) [inline src/builder/ModuleHelper.js]: I think the name `installer` is very confusing. what about `descriptor` ?

Comment (kptdobe) [reply src/builder/ModuleHelper.js]: I knew you will find a better name for it. I could not :)

File: src/build.js (hunk 1 above)
Comment (kptdobe) [reply src/build.js]: I saw that `up` extends `build` thus I thought I do not it here.
`deploy` does not seem to run a `build` thus we would not need it here.
`package` needs it, correct, I miss it.
I'll add the arg to `yargsBuild`.

Comment (kptdobe) [reply src/build.js]: ok, found it. `deploy` calls `package` which calls `build`...

**Comment by kptdobe** [comment]
@tripodsan I added support for the option in `build`, `up`, `package` and `deploy`, together with corresponding tests.
I do not understand though the codecov drop...

**Comment by kptdobe** [comment]
> I think we might revisit the way we collect common argument...it's kind of stupid to add the same stuff to up,build,deploy,package....

I agree, the parameter handling could be generalised.

**Pull request merged**